resources/qml/Sidebar.qml

```diff
@@ -166,15 +166,6 @@ Rectangle
         anchors.topMargin: UM.Theme.getSize("default_margin").height
     }
 
-    currentModeIndex:
-    {
-        var index = parseInt(UM.Preferences.getValue("cura/active_mode"))
-        if(index)
-        {
-            return index;
-        }
-        return 0;
-    }
```

Comment:
Yes. The important bit of this code is moved to Component.onCompleted.
The problem I'm fixing here is that currentModeIndex was being set before the modesListModel was populated.